brion/blueConfig.cpp

```diff
@@ -137,18 +147,69 @@ class BlueConfig
 
                 std::string value = line.substr( pos+1 );
                 boost::trim( value );
-                _table[type][name].insert( std::make_pair( line.substr( 0, pos),
+                table[type][name].insert( std::make_pair( line.substr( 0, pos),
                                                            value ));
             }
         }
 
-        if( _table[CONFIGSECTION_RUN].empty( ))
+        if( table[CONFIGSECTION_RUN].empty( ))
             LBTHROW( std::runtime_error( source +
                                          " not a valid BlueConfig file" ));
     }
 
-    Strings _names[CONFIGSECTION_ALL];
-    ValueTable _table[CONFIGSECTION_ALL];
+    std::string getRun()
+    {
+        const brion::Strings& runs = names[ brion::CONFIGSECTION_RUN ];
+        return runs.empty() ? std::string() : runs.front();
+    }
+
+    const std::string& get( const BlueConfigSection section,
+                            const std::string& sectionName,
+                            const std::string& key  ) const
+    {
+        // This function doesn't create entries in the tables in case they
+        // don't exist.
+        static std::string empty;
+        const ValueTable::const_iterator tableIt =
+            table[section].find( sectionName );
+        if( tableIt == table[section].end( ))
+            return empty;
+        const KVStore& store = tableIt->second;
+        const KVStore::const_iterator kv = store.find( key );
+        if( kv == store.end( ))
+            return empty;
+        return kv->second;
+    }
+
+    const std::string& getCircuitTarget()
+    {
+        return get( brion::CONFIGSECTION_RUN, getRun(), "CircuitTarget" );
```

Comment:
On one hand, we use a define for "Run" section, but for an explicit entry it's a magic string??